plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/bedrock-runtime.ts

```diff
@@ -579,4 +590,145 @@ export class BedrockRuntimeServiceExtension implements ServiceExtension {
       }
     }
   }
+
+  private async responseHookInvokeModelWithResponseStream(
+    response: NormalizedResponse,
+    span: Span,
+    tracer: Tracer,
+    config: AwsSdkInstrumentationConfig
+  ): Promise<any> {
+    const stream = response.data?.body;
+    const modelId = response.request.commandInput?.modelId;
+    if (!stream) return;
+
+    const wrappedStream =
+      BedrockRuntimeServiceExtension.instrumentAsyncIterable(
+        stream,
+        async (chunk: { chunk?: { bytes?: Uint8Array } }) => {
+          const parsedChunk = BedrockRuntimeServiceExtension.parseChunk(
+            chunk?.chunk?.bytes
+          );
+
+          if (!parsedChunk) return;
+
+          if (modelId.includes('amazon.titan')) {
+            BedrockRuntimeServiceExtension.recordTitanAttributes(
+              parsedChunk,
+              span
+            );
+          } else if (modelId.includes('anthropic.claude')) {
+            BedrockRuntimeServiceExtension.recordClaudeAttributes(
+              parsedChunk,
+              span
+            );
+          } else if (modelId.includes('amazon.nova')) {
+            BedrockRuntimeServiceExtension.recordNovaAttributes(
+              parsedChunk,
+              span
+            );
+          }
+        }
+      );
+    // Replace the original response body with our instrumented stream.
+    // - Defers span.end() until the entire stream is consumed
+    // This ensures downstream consumers still receive the full stream correctly,
+    // while OpenTelemetry can record span attributes from streamed data.
+    response.data.body = (async function* () {
+      try {
+        for await (const item of wrappedStream) {
+          yield item;
+        }
+      } finally {
+        span.end();
+      }
+    })();
+    return response.data;
+  }
+  // Tap into the stream at the chunk level without modifying the chunk itself.
+  private static instrumentAsyncIterable<T>(
+    stream: AsyncIterable<T>,
+    onChunk: (chunk: T) => void
+  ): AsyncIterable<T> {
+    return {
+      [Symbol.asyncIterator]: async function* () {
+        for await (const chunk of stream) {
+          onChunk(chunk);
+          yield chunk;
+        }
+      },
+    };
+  }
+
+  private static parseChunk(bytes?: Uint8Array): any {
+    if (!bytes || !(bytes instanceof Uint8Array)) return null;
+    try {
+      const str = Buffer.from(bytes).toString('utf-8');
+      return JSON.parse(str);
+    } catch (err) {
+      console.warn('Failed to parse streamed chunk', err);
```

Comment:
Shouldn't use `console` in instrumentation code. Instead use the DiagLogger that every intsrumentation instance has.

```suggestion
      this._diag.warn('Failed to parse streamed chunk', err);
```

This will mean you'll need to not use `static` for this method.

Reply:
Thank you, @trentm , for the review . I addressed all your comments and would appreciate next round of a review. Thanks